packages/kbn-expandable-flyout/src/components/preview_section.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiButtonEmpty, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import React from 'react';
+import { css } from '@emotion/react';
+import {
+  PREVIEW_SECTION,
+  PREVIEW_SECTION_BACK_BUTTON,
+  PREVIEW_SECTION_CLOSE_BUTTON,
+} from './test_ids';
+import { useExpandableFlyoutContext } from '../..';
+import { BACK_BUTTON } from './translations';
+
+interface PreviewSectionProps {
+  /**
+   * Component to be rendered
+   */
+  component: React.ReactElement;
+  /**
+   * Width used when rendering the panel
+   */
+  width: number | undefined;
+  /**
+   * Display the back button in the header
+   */
+  showBackButton: boolean;
+}
+
+/**
+ * Preview section of the expanded flyout rendering one or multiple panels.
+ * Will display a back and close button in the header for the previous and close feature respectively.
+ */
+export const PreviewSection: React.FC<PreviewSectionProps> = ({
+  component,
+  showBackButton,
+  width,
+}: PreviewSectionProps) => {
+  const { closePreviewPanel, previousPreviewPanel } = useExpandableFlyoutContext();
+
+  const previewWith: string = width ? `${width}px` : '0px';
```

Comment:
idk how this related but kibana generally uses this sort of thing instead of expressing sizes directly, maybe its worth checking out 
![image](https://user-images.githubusercontent.com/11671118/217272073-4331cbf5-c945-4b03-b2d7-a8f8d36399d9.png)

Reply:
good idea, made the change for the margin, thanks!